runtime/http_client.go

```diff
@@ -32,6 +36,17 @@ type HTTPClient struct {
 	BaseURL string
 }
 
+// RawHTTPError defines an error for HTTP
+type RawHTTPError struct {
+	StatusCode int
+	RawBody    []byte
+}
+
+func (rawErr *RawHTTPError) Error() string {
+	return "Unexpected http client response (" +
+		strconv.Itoa(rawErr.StatusCode) + ")"
```

Comment:
Sure can rename it.

Rather not put the entire body in the in the error message as it's just raw bytes, too easy to return the error back out which leads to double encoded JSON.